### PR TITLE
feat: gutenberg custom editor blocks for clip paths

### PR DESCRIPTION
**Pull request by zoracreates**
This solves the clip path filters  #102 and also the h1 clip path stylings #79.

As a caveat, already inserted Page headings (h2 - h4) and buttons must be updated through the editor so that they have the new c4aa-clipPath classes. I can do this later on backtobasics but you'll have to make the updates locally.

**Updating Buttons**
Screen Shot 2019-09-19 at 7 36 21 PM

------
------
------

**Updating Headings**

Screen Shot 2019-09-19 at 7 49 34 PM

### CODE REVIEW

**Comment by zoracreates** [comment]
I also styled the newsletter page/form per #26

File: css/c4aa-clipPaths.css [hunk 1]
```diff
@@ -39,82 +36,93 @@ legend {
...
 }
 
-.c4aa-default-button-wrapper:hover {
+.is-style-c4aa-clipPath-button:hover {
 	filter: none;
 }
 
-.c4aa-outline-button-wrapper  {
+.entry .entry-content .wp-block-button.is-style-c4aa-clipPath-button-outline .wp-block-button__link:not(.has-text-color),
+.entry .entry-content .wp-block-button.is-style-c4aa-clipPath-button-outline .wp-block-button__link:not(.has-text-color):hover,
```
Comment (laras126) [inline css/c4aa-clipPaths.css]: Oh man, this makes me think we should totally have put Sass in here...

File: js/editor.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+wp.domReady( () => {
+    
+	wp.blocks.unregisterBlockStyle( 'core/button', 'outline' );
+    wp.blocks.unregisterBlockStyle( 'core/button', 'squared' );
```
Comment (laras126) [inline js/editor.js]: Looks like some whitespace issues - can you update @zoracreates ?

File: js/script.js [hunk 3]
```diff
@@ -1,77 +1,109 @@
...
+// h1 elements without background image, and archive h2 titles
+const entryTitles = document.querySelectorAll( 'header:not(.featured-image) h1.entry-title, .archive h2.entry-title' );
 
-/* Button Clip Paths 
+// widget titles
+const widgetTitles = document.querySelectorAll('.widget-title');
 
-	because clip-paths eliminates drop shadows and borders, 
-	wrappers are added to buttons to contain these properties
+//donate legent titles
```
Comment (laras126) [inline js/script.js]: Typo

**Comment by zoracreates** [comment]
>One thought though - do we need to resave all posts to update the block markup in the database? That is one caveat about Gutenberg I forgot about...

@laras126 
I think so. This I think it shouldn't take that long for the individual/higher-level pages like Home, About, Donate, etc. 

As for the News posts (which are a lot), we could set some sort of JS fallback for them. 

I'd have to test this, but this should do the trick:
```
// select news post headings with no c4aa-clipPath class

const newsHeadings =  document.querySelectorAll(  '.type-post h2:not(.is-style-c4aa-clipPath-heading):not(.entry-title),  .type-post h3:not(.is-style-c4aa-clipPath-heading), .type-post h4:not(.is-style-c4aa-clipPath-heading)' )

addHeadingClipPaths(newsHeading);

```

It doesn't look like these posts have any buttons, and if they do then they will just get a default red button with a drop shadow and no clip-path.

**Pull request merged**